var/spack/repos/builtin/packages/vtkm/package.py

```diff
@@ -29,15 +31,15 @@ class Vtkm(CMakePackage, CudaPackage):
     # can overwhelm compilers with too many symbols
     variant('build_type', default='Release', description='CMake build type',
             values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
-    variant("shared", default=True, description="build shared libs")
+    variant("shared", default=False, description="build shared libs")
```

Comment:
Why default `shared` to `false`?

Reply:
Because when CUDA is enabled we don't support shared builds as we require CUDA separable compilation.

This also now matches VTK-m CMake defaults which is static

Reply:
If static will be particularly common and often preferred then there should probably be a `pic` variant to express "static + PIC" that's on by default and conflicts with shared.

Reply:
static builds of VTK-m are always +PIC so that VTK-m can be embedded in a consuming library

Reply:
It should still be there for consistency in how that's expressed in spack.  If a package supports static and shared then the convention getting settled on is to have `+shared~pic`, `~shared+pic`, and `~shared~pic` for shared, static with pic, and static without pic  configurations respectively.  If there is no valid "static without pic" configuration then you can add a `conflicts("~shared", when="~pic")`

Reply:
Thanks for the explanation. I have updated the package, does this look better @chuckatkins